### PR TITLE
For #27627 - Check the TCP CFR status lazily

### LINKED ISSUE #27627: Ensure the TCP CFR is correctly enabled from Nimbus
@cpeterso reported an issue with the TCP CFR and on further investigations it seems like the setting for is has a bug stemming from a race between the setting initialization and the Nimbus initialization.
The other MR experiments are not affected because they use lazy initialization - something that the TCP CFR could use also.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21901)

### PR DESCRIPTION
**Pull request by Mugurell**
Use lazyFeatureFlagPreference as a quick small way to avoid a race between initializing the value and the Nimbus initialization based on which the value should be calculated.
This is the same flow that the other MR experiments use.

Video showing the CFR correctly shown for this preview experiment - https://experimenter.services.mozilla.com/nimbus/firefox-android-tcp-cfr-experiment:

https://user-images.githubusercontent.com/11428869/199061908-3eb150f5-7c0e-4029-b359-b89e7825122a.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27627

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 1]
```diff
@@ -610,9 +610,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     /**
      * Indicates if the total cookie protection CRF should be shown.
      */
-    var shouldShowTotalCookieProtectionCFR by booleanPreference(
+    var shouldShowTotalCookieProtectionCFR by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_should_show_total_cookie_protection_popup),
-        default = mr2022Sections[Mr2022Section.TCP_CFR] == true,
+        featureFlag = true,
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:615]: Not pretty but using a `lazyFeatureFlagPreference` means a quick (will have to be uplifted) and safe (used by the other MR features) general approach.

**Review by Amejia481 — APPROVED** [review]

**Comment by cpeterso** [comment]
@Mugurell After your bug fix, when a user enrolls in the `treatment-a` branch, will the user see the TCP CFR on their very next page load? Or will they need to kill and restart Fenix before the TCP CFR is displayed?

**Comment by Mugurell** [comment]
> @Mugurell After your bug fix, when a user enrolls in the `treatment-a` branch, will the user see the TCP CFR on their very next page load? Or will they need to kill and restart Fenix before the TCP CFR is displayed?

The app will have to be restarted. Same as with the other MR experiments.

**Comment by Mugurell** [comment]
@Mergifyio backport releases_v107.0.0

**Pull request merged**